### PR TITLE
Support BC7

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This PR adds support for BC7 and related new features.

#### New APIs:
- `SubscriptionOption.installmentInfo` which contains information about installment plans you might want to display in your paywall.
- `PurchasesConfiguration.pendingTransactionsForPrepaidPlansEnabled` allows to enable pending prepaid subscriptions.

#### Breaking changes
- Increase min Sdk to 21. This was increased in BC7
- Remove `UpgradeInfo`
- Remove all `purchasePackage` and `purchaseProduct` methods using `UpgradeInfo`. If you want to perform upgrades/downgrades, please migrate to use the `purchase(PurchaseParams)` method, which has options to perform that operation.

### CODE REVIEW

**Review by joshdholtz — APPROVED** [review]
:shipit:

File: purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt [hunk 1]
```diff
@@ -277,6 +277,7 @@ class Purchases internal constructor(
             val configuration = PurchasesConfiguration.Builder(context, apiKey)
                 .appUserID(appUserID)
                 .dangerousSettings(DangerousSettings(customEntitlementComputation = true))
+                .pendingTransactionsForPrepaidPlansEnabled(true)
```
Comment (vegaro) [inline purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt:280]: why is this always true?

Comment (tonidero) [reply purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt:280]: I believe we want to enable it by default for companies using `customEntitlementComputation`. We can provide an option to disable it later if needed though, but we try to keep the API very small in the `customEntitlementComputation` flavor I think.

File: README.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ Or view / build our Android sample app:
 ## Requirements
 - Java 8+
 - Kotlin 1.5.0+
```
[the remark this answers is not in the document]
Comment (tonidero) [reply README.md:53]: Ohh good catch! Will update that since I'm writing the migration docs.

**Pull request merged**